### PR TITLE
fix(map,security): admin + map code review fixes

### PR DESCRIPTION
**Pull request by jansinger**
## Summary

Two code review passes — admin area + sighting map.

### Admin Fixes (first commit)

- **🔴 Critical:** Weather refresh API `/api/admin/weather/[id]/refresh` was publicly accessible — missing `requireUserRole` call (all other admin APIs had it, this was an oversight)
- **🟡 DaisyUI v5:** Remove deprecated `modal-open` class from `MediaModal.svelte` (v4 pattern — v5 uses native `<dialog>.showModal()`)
- **🟡 Date validation:** Add `isValidDateParam()` with `YYYY-MM-DD` regex for admin date filter params before passing to SQL
- **🟡 Drizzle type safety:** Replace `Record<string, unknown>` with typed interface in approve endpoint; fix `approvedAt` from ISO string to `Date` object (schema expects `timestamp({ mode: 'date' })`)
- **🟡 Bonus bug:** `error(400)` inside try/catch was re-thrown as 500 because `HttpError` is not `instanceof Error` — fixed with `isHttpError()` from `@sveltejs/kit`

### Map Fixes (second commit)

- **🔴 Fix A:** Geolocation event listeners never removed — store `EventsKey[]`, call `unByKey` + `geolocation.dispose()` in `dispose()`
- **🔴 Fix B:** View `change:resolution` listener key stored + `unByKey` in `dispose()`
- **🔴 Fix C:** Floating `setYear()` promises add `.catch()` — prevents unhandled rejections
- **🟡 Fix D:** Replace hardcoded 1500ms init timeout with real loading signal (`firstLoadComplete` flag in `onLoading` callback) — removes race condition
- **🟡 Fix E:** Add `aria-label` to `LocationControl` + `ZoomAllControl` buttons (WCAG 2.1 AA)
- **🟡 Fix F:** Export `clearStyleCache()` from `styleUtils.ts` and call in `dispose()` to free cached styles
- **🟡 Fix G:** Remove dead code — `addFeatureFromCoordinate()`, unused `+page.server.ts` (mapConfig never consumed by `+page.svelte`), unused `MapDataLoader` class + tests

## Test plan

- [x] `npm run test:quick` — 1309 tests passing, no regressions
- [x] `clearStyleCache()` tested in `styleUtils.test.ts` (RED → GREEN confirmed)
- [x] TDD: RED → GREEN confirmed for admin fixes
- [x] lint + type-check + svelte-check: 0 errors

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by jansinger — COMMENTED** [review]
Alle 4 Review-Kommentare adressiert (commit 4beb692):

- **dateParam**: isExistingCalendarDate() ergänzt — lehnt 2024-02-31, 2023-02-29 etc. ab (UTC Date round-trip); Tests für nicht-existente Daten und Schaltjahr-Edge-Cases hinzugefügt
- **+page.server.ts**: Folgt automatisch aus dem dateParam-Fix
- **approve GET**: instanceof Error && 'status' in err → isHttpError() — konsistent mit PATCH; Test für 404-Propagation hinzugefügt  
- **weather refresh 500**: error.message aus Response entfernt, generische Meldung; Test prüft dass keine sensiblen Daten in der Response landen

File: src/lib/map/optimizedMapController.ts [hunk 1]
```diff
@@ -253,7 +257,10 @@ export class SichtungenMap {
 		}
 
 		// Lade Daten für das aktuelle Jahr
-		this.setYear(this.displayedYear);
+		void this.setYear(this.displayedYear).catch((err) => {
+			logger.error({ err }, 'Initial sightings load failed');
+			this.loadingCallback?.(false);
+		});
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/map/optimizedMapController.ts:269]: Umgesetzt in commit 50e0531: `onError?: (error: Error) => void` zu `MapOptions` hinzugefügt, `errorCallback` als private Feld gespeichert. Wird jetzt aus allen drei Fehlerpfaden aufgerufen: initial setYear, Jahr-Wechsel-Event und applyFilter (außer AbortError).

File: src/lib/components/map/SightingsMapView.svelte [hunk 2]
```diff
@@ -136,6 +134,11 @@
 					if (loading) {
 						loadingType = 'features';
 						errorMessage = null;
+					} else if (!firstLoadComplete) {
+						firstLoadComplete = true;
+						countManager.updateCounts();
+						currentDisplayedYear = mapInstance!.getDisplayedYear();
+						isInitialLoading = false;
 					}
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/lib/components/map/SightingsMapView.svelte:142]: Umgesetzt in commit 50e0531: Der neue `onError`-Callback des Controllers wird in `SightingsMapView` an `onError: (err) => { errorMessage = '...'; isLoadingData = false; isInitialLoading = false; }` gebunden. Der Error-Toast ist damit wieder erreichbar — ohne den globalen unhandledrejection-Handler, der alle Promise-Fehler der Seite abgefangen hat.

File: src/routes/admin/+page.server.ts [hunk 3]
```diff
@@ -22,8 +24,8 @@ export const load: PageServerLoad = async ({ url }) => {
 	// Bedingungen für die SQL-Abfrage sammeln
 	const conditions = [];
 
-	// Datums-Filter
-	if (dateFrom && dateTo) {
+	// Datums-Filter (nur mit validiertem YYYY-MM-DD Format)
+	if (isValidDateParam(dateFrom) && isValidDateParam(dateTo)) {
 		conditions.push(sql`${sightings.sightingDate} BETWEEN ${dateFrom} AND ${dateTo}`);
```
[the remark this answers is not in the document]
Comment (jansinger) [reply src/routes/admin/+page.server.ts]: Umgesetzt in commit 50e0531: `DATE(${sightings.sightingDate}) BETWEEN` statt direktem Timestamp-Vergleich. Dadurch wird der gesamte `dateTo`-Tag inkludiert, unabhängig von der Uhrzeit des Eintrags.
